storage/spec/files_spec.rb

```diff
@@ -108,6 +121,26 @@ def capture &block
     expect(storage_file_content "file.txt").to eq "Content of test file.txt\n"
   end
 
+  it "can upload a local file to a bucket with encryption key" do
+    delete_file "enc_file.txt"
+    expect(@bucket.file "enc_file.txt").to be nil
+
+    expect {
+      upload_encrypted_file(project_id:            @project_id,
+                            bucket_name:           @bucket_name,
+                            local_file_path:       @local_enc_file_path,
+                            storage_file_path:     "enc_file.txt",
+                            base64_encryption_key: @encoded_enc_key)
+    }.to output(
+      "Uploaded enc_file.txt with encryption key\n"
+    ).to_stdout
+
```

Comment:
What happens if you try to download the file without providing an encryption key?

I noticed that the upload & download tests are exactly the same as the ones that don't use customer-supplied encryption keys.  This is great but it make me wonder: *Is there a way that we could verify the file was really encrypted?*

Not required, but I'm curious what happens.  If you download a file *without* passing the key, is an error thrown, or is the file successfully downloaded (but the content is encrypted)?  If, for example, an error is thrown, it might make the test more clear if:

 1. Upload w/ encryption key
 1. Try download, verify error thrown
 1. Download, verify content OK

Just a thought!